src/lib/serviceRoutinesV2/getEntities.cpp

```diff
@@ -61,17 +60,58 @@ std::string getEntities
   ParseData*                 parseDataP
 )
 {
-  std::string  answer;
-  Entities     entities;
+  using namespace std;
+  typedef vector<string>  vstr;
+  typedef vstr::size_type vstr_sz;
+
+  Entities entities;
+  string   answer;
+
+  // optional parameter for list of IDs
+  string paramIDs     = ciP->uriParam["id"];
+  string paramPattern = ciP->uriParam["idPattern"];
+  string pattern      = ".*"; // all entities, default value
 
+  if (paramPattern != "" && paramIDs != "")
```

Comment:
Parenthesis makes the condition easier to read:

    ((paramPattern != "") && (paramIDs != ""))